source/agora/common/TransactionData.d

```diff
@@ -0,0 +1,284 @@
+/*******************************************************************************
+
+    Contains a storage class consisting of a maximum size fixed byte array.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.TransactionData;
+
+import agora.common.Hash;
+import agora.common.Serializer;
+
+static import std.ascii;
+import std.conv: parse;
+import std.algorithm.iteration : each, map;
+import std.range;
+import std.utf;
+
+
+public struct TransactionData
+{
+    invariant()
+    {
+        assert(length <= MAX_SIZE);
+    }
+
+    /// The maximum size
+    public static immutable MAX_SIZE = 1024;
+
+    /// The byte array of transaction data
+    public ubyte[MAX_SIZE] data;
```

Comment:
This means the data will always be 1kb in memory, even for 300 bytes. That's not desireable.